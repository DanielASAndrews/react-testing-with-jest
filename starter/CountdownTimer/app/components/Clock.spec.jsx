import React from 'react';
import { mount, shallow } from 'enzyme';
import Clock from 'Clock';

describe('Clock', () => {
    it('should render Clock without crashing', () => {
        mount(<Clock />);
    });

    describe('render', () => {
        it('should render the clock', () => {
            const clock = shallow(<Clock timeInSeconds={60} />);
            const time = <span className="clock-text">01:00</span>;

            expect(clock.contains(time)).toEqual(true);
            expect(clock.find('.clock-text').text()).toBe("01:00");
        });
    });

    describe('formatTime', () => {
        it('should format time', ()=> {
            const clock = shallow (<Clock />);
            const seconds = 120;
            const expected = "02:00";
            const actual = clock.instance().formatTime(seconds);

            expect(actual).toBe(expected);
        });

        it('should format seconds when minutes or seconds are less than 10', ()=>{
            const clock = shallow(<Clock />);
            const seconds = 9;
            const expected = "00:09";
            const actual = clock.instance().formatTime(seconds);

            expect(actual).toBe(expected);
        });
    });
});
