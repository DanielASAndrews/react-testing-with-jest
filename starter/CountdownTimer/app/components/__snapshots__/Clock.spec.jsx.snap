// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock Snapshots formatTime should format seconds when minutes or seconds are less than 10 1`] = `"00:09"`;

exports[`Clock Snapshots formatTime should format time 1`] = `"02:00"`;

exports[`Clock Snapshots render should render the clock 1`] = `
<div
  className="clock"
>
  <span
    className="clock-text"
  >
    01:00
  </span>
</div>
`;

exports[`Clock Snapshots should render Clock without crashing 1`] = `
<Clock
  timeInSeconds={0}
>
  <div
    className="clock"
  >
    <span
      className="clock-text"
    >
      00:00
    </span>
  </div>
</Clock>
`;
